vllm/executor/ray_gpu_executor.py

```diff
@@ -161,6 +161,10 @@ def _init_workers_ray(self, placement_group: "PlacementGroup",
         self._run_workers("update_environment_variables",
                           all_args=all_args_to_update_environment_variables)
 
+        if len(node_gpus) == 1:
+            # in single node case, we don't need to get the IP address.
+            # localhost is sufficient
+            driver_ip = "127.0.0.1"
         distributed_init_method = get_distributed_init_method(
             driver_ip, get_open_port())
```

Comment:
do we need to change this for the ray case? the reason why it's bad is the logic above changes behavior based on the driver_ip. so it can make for confusing debugging if this logic assumes certain things about the driver ip, but then we change it.

Reply:
it is needed as reported in #3466 . for example, if a node has two network interface, one has loopback address 127.0.0.1 , and another has address for communication in the subnet 172.16.0.2 . And if `ray` gets the `172.16.0.2`, processes cannot communicate via this ip. See https://buildkite.com/vllm/ci-aws/builds/1919#01902ef5-264a-4d1f-b10f-c6917efea505 for a realistic failure.

Anyway, using the loopback address seems a risk-free solution for me. Because we know this is a single-node environment, and `127.0.0.1` is always a loopback address.

Reply:
got it. can you note this as a comment? `hardcode driver ip to loopback in case node has multiple nics`

Reply:
fixed in [cf3840c](https://github.com/vllm-project/vllm/pull/5619/commits/cf3840cb9e4a45bef98ddfc339c2c36560a52731)